src/Cache/CacheClient.php

```diff
@@ -48,11 +48,11 @@ class CacheClient implements LoggerAwareInterface
 
     /**
      * @param IConfiguration $configuration
-     * @param ICredentialProvider $authProvider
+     * @param CredentialProvider $authProvider
      * @param int $defaultTtlSeconds
      */
     public function __construct(
-        IConfiguration $configuration, ICredentialProvider $authProvider, int $defaultTtlSeconds
+        IConfiguration $configuration, CredentialProvider $authProvider, int $defaultTtlSeconds
```

Comment:
```suggestion
        IConfiguration $configuration, ICredentialProvider $authProvider, int $defaultTtlSeconds
```

Reply:
Sorry about the runaround, but we'll want these types to be `ICredentialProvider` pretty much everywhere in the clients and grpc managers.